test/Spec.hs

```diff
@@ -908,6 +908,97 @@ main =
       it "use add" $ ruleCatches useAdd "COPY packaged-app.tar /usr/src/app"
       it "use not add" $ ruleCatchesNot useAdd "COPY package.json /usr/src/app"
     --
+    describe "`COPY` without `WORKDIR` set" $ do
+      it "ok: `COPY` with absolute destination and no `WORKDIR` set" $ do
+        ruleCatchesNot relativeCopyWithoutWorkdir "COPY bla.sh /usr/local/bin/blubb.sh"
+        onBuildRuleCatchesNot relativeCopyWithoutWorkdir "COPY bla.sh /usr/local/bin/blubb.sh"
+      it "ok: `COPY` with absolute destination and no `WORKDIR` set - windows" $ do
+        ruleCatchesNot relativeCopyWithoutWorkdir "COPY bla.sh c:\\system32\\blubb.sh"
+        onBuildRuleCatchesNot relativeCopyWithoutWorkdir "COPY bla.sh d:\\mypath\\blubb.sh"
+      it "ok: `COPY` with relative destination and `WORKDIR` set" $ do
+        ruleCatchesNot relativeCopyWithoutWorkdir "WORKDIR /usr\nCOPY bla.sh blubb.sh"
+        onBuildRuleCatchesNot relativeCopyWithoutWorkdir "WORKDIR /usr\nCOPY bla.sh blubb.sh"
+      it "ok: `COPY` with relative destination and `WORKDIR` set - windows" $ do
+        ruleCatchesNot relativeCopyWithoutWorkdir "WORKDIR c:\\system32\nCOPY bla.sh blubb.sh"
+        onBuildRuleCatchesNot relativeCopyWithoutWorkdir "WORKDIR c:\\system32\nCOPY bla.sh blubb.sh"
+      it "not ok: `COPY` with relative destination and no `WORKDIR` set" $ do
+        ruleCatches relativeCopyWithoutWorkdir "COPY bla.sh blubb.sh"
+        onBuildRuleCatches relativeCopyWithoutWorkdir "COPY bla.sh blubb.sh"
+      it "not ok: `COPY` to relative destination if `WORKDIR` is set in a previous stage but not inherited" $
+        let dockerFile =
+              Text.unlines
+                [ "FROM debian:buster as stage1",
+                  "WORKDIR /usr",
+                  "FROM debian:buster",
+                  "COPY foo bar"
+                ]
+        in do
+          ruleCatches relativeCopyWithoutWorkdir dockerFile
+          onBuildRuleCatches relativeCopyWithoutWorkdir dockerFile
+      it "not ok: `COPY` to relative destination if `WORKDIR` is set in a previous stage but not inherited - windows" $
+        let dockerFile =
+              Text.unlines
+                [ "FROM microsoft/windowsservercore as stage1",
+                  "WORKDIR c:\\system32",
+                  "FROM microsoft/windowsservercore",
+                  "COPY foo bar"
+                ]
+        in do
+          ruleCatches relativeCopyWithoutWorkdir dockerFile
+          onBuildRuleCatches relativeCopyWithoutWorkdir dockerFile
+      it "ok: `COPY` to relative destination if `WORKDIR` has been set in base image" $
+        let dockerFile =
+              Text.unlines
+                [ "FROM debian:buster as base",
+                  "WORKDIR /usr",
+                  "FROM debian:buster as stage-inbetween",
```

Comment:
This case seems buggy. In order to inherit from a previous stage you need here:

```dockerfile
FROM base as stage-inbetween
```

This example should trigger the rule

Reply:
oh, my bad. I misread the test case